aptos-move/block-executor/src/executor.rs

```diff
@@ -446,6 +449,7 @@ where
         versioned_cache: &MVHashMap<T::Key, T::Value, X>,
         last_input_output: &TxnLastInputOutput<T::Key, E::Output, E::Error>,
         base_view: &S,
+        txn_commit_listener: &Option<L>,
```

Comment:
this probably should be hold in the BlockExecutor instead of passing all the way down

Reply:
Can you clarify what you have in mind? Is this by adding a callback in TransactionOutput function?

Reply:
initially I thought we can hold this listener in the block executor, but in the end I was wondering about supporting it directly in the TransactionOutput trait. if we could add the `on_commit` function to the trait so that we can directly call `self.output[txn_idx].on_commit()` in the commit_hook().

when we construct AptosTransactionOutput::new, we can pass in Option<Listener> that can be called later in on_commit, we don't need to have the no-op one, it can just be None.

Reply:
Hmm, we can do that but not sure if the API will be cleaner that way. basically we need to add something like `on_committed(&self, txn_idx: TxnIndex, execution_status: &Self::ExecutionStatus)` to the `TransactionOutput` trait and the `txn_idx` and the `execution_status` seems out of place to be in the API.

Reply:
As discussed offline, renamed the lister to commit hook interface but not adding it to the `TransactionOutput` trait as that makes the trait unclean.